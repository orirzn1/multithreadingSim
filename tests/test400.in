L13
S6
O11
N10

T0
I@0x00000000
LOAD $2, $4, $2
STORE $1, $5, $3
LOAD $2, $4, 0x3
SUBI $6, $1, 0x3
STORE $6, $5, $4
STORE $0, $3, 0x3
STORE $6, $1, 0x3
ADDI $0, $4, 0x2
STORE $1, $1, 0x1
STORE $7, $7, $5
HALT $0

T1
I@0x00000000
ADDI $2, $2, 0x1
SUBI $2, $7, 0x4
SUBI $1, $0, 0x3
ADD $3, $1, $7
ADD $0, $7, $3
SUBI $1, $0, 0x1
HALT $0

T2
I@0x00000000
SUB $5, $6, $6
SUB $6, $5, $7
ADD $1, $2, $4
ADD $6, $0, $7
LOAD $6, $0, $6
ADDI $4, $6, 0x4
ADD $3, $0, $1
HALT $0

T3
I@0x00000000
STORE $0, $5, 0x3
SUB $0, $6, $5
LOAD $7, $3, 0x0
LOAD $7, $7, $2
SUB $4, $4, $6
LOAD $1, $3, 0x2
ADDI $0, $5, 0x4
SUB $4, $6, $2
ADDI $6, $5, 0x2
HALT $0

T4
I@0x00000000
ADD $5, $0, $4
STORE $2, $4, $5
LOAD $1, $5, $0
STORE $5, $2, $4
LOAD $4, $3, $1
SUB $1, $3, $0
LOAD $1, $4, $2
HALT $0

T5
I@0x00000000
SUB $1, $3, $3
LOAD $0, $1, $1
ADDI $6, $1, 0x4
SUB $3, $1, $4
HALT $0

T6
I@0x00000000
SUB $6, $0, $1
SUB $0, $3, $5
STORE $0, $2, $0
SUB $4, $2, $7
ADDI $7, $7, 0x2
LOAD $0, $1, $5
HALT $0

T7
I@0x00000000
SUB $0, $5, $1
STORE $7, $2, 0x2
STORE $0, $2, 0x4
STORE $7, $7, $5
ADD $5, $2, $6
ADD $0, $3, $7
ADDI $7, $3, 0x3
ADD $3, $1, $6
HALT $0

T8
I@0x00000000
ADDI $4, $0, 0x2
SUB $4, $5, $1
SUBI $5, $1, 0x1
SUBI $3, $2, 0x4
ADD $1, $2, $2
ADD $2, $1, $2
HALT $0

T9
I@0x00000000
LOAD $6, $7, 0x4
SUBI $6, $1, 0x3
SUB $4, $4, $3
ADD $0, $4, $7
SUBI $1, $4, 0x4
LOAD $3, $7, 0x2
HALT $0

D@0x0
0x6
0x2
0x3
0x2
0x4
0xA
