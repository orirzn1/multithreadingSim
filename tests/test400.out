
---- Blocked MT Simulation ----

Register file thread id 0:
	R0 = 0x2	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFD	R7 = 0x0
Register file thread id 1:
	R0 = 0xFFFFFFFD	R1 = 0xFFFFFFFC	R2 = 0xFFFFFFFC	R3 = 0xFFFFFFFD	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0xA	R5 = 0x0	R6 = 0x6	R7 = 0x0
Register file thread id 3:
	R0 = 0x4	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x2	R7 = 0x0
Register file thread id 4:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 5:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x4	R7 = 0x0
Register file thread id 6:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x2
Register file thread id 7:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x3
Register file thread id 8:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0xFFFFFFFC	R4 = 0x0	R5 = 0xFFFFFFFF	R6 = 0x0	R7 = 0x0
Register file thread id 9:
	R0 = 0x0	R1 = 0xFFFFFFFC	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFD	R7 = 0x0
Blocked MT CPI for this program 5.759494

-----Finegrained MT Simulation -----

Register file thread id 0:
	R0 = 0x2	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFD	R7 = 0x0
Register file thread id 1:
	R0 = 0xFFFFFFFD	R1 = 0xFFFFFFFC	R2 = 0xFFFFFFFC	R3 = 0xFFFFFFFD	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x4	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 3:
	R0 = 0x4	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x2	R7 = 0x0
Register file thread id 4:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 5:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x4	R7 = 0x0
Register file thread id 6:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x2
Register file thread id 7:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x3
Register file thread id 8:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0xFFFFFFFC	R4 = 0x0	R5 = 0xFFFFFFFF	R6 = 0x0	R7 = 0x0
Register file thread id 9:
	R0 = 0x0	R1 = 0xFFFFFFFC	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFD	R7 = 0x0
Finegrained Multithreading CPI for this program 1.164557

